api/src/controllers/extension-permissions.ts

```diff
@@ -0,0 +1,222 @@
+import { isDirectusError } from '@directus/errors';
+import express from 'express';
+import { ErrorCode, RouteNotFoundError } from '../errors/index.js';
+import { respond } from '../middleware/respond.js';
+import { validateBatch } from '../middleware/validate-batch.js';
+import { ExtensionPermissionsService } from '../extensions/service.js';
+import { MetaService } from '../services/meta.js';
+import type { PrimaryKey } from '../types/index.js';
+import asyncHandler from '../utils/async-handler.js';
+import { sanitizeQuery } from '../utils/sanitize-query.js';
+import useCollection from '../middleware/use-collection.js';
+import { getExtensionManager } from '../extensions/extensions.js';
+
+const router = express.Router();
+
+router.use(useCollection('directus_extension_permissions'));
+
+router.post(
+	'/',
+	asyncHandler(async (req, res, next) => {
+		if (req.accountability?.admin !== true) throw new RouteNotFoundError({ path: req.path });
+
+		const service = new ExtensionPermissionsService({
+			accountability: req.accountability,
+			schema: req.schema,
+		});
+
+		const savedKeys: PrimaryKey[] = [];
+
+		if (Array.isArray(req.body)) {
+			const keys = await service.createMany(req.body);
+			savedKeys.push(...keys);
+		} else {
+			const key = await service.createOne(req.body);
+			savedKeys.push(key);
+		}
+
+		try {
+			if (Array.isArray(req.body)) {
+				const items = await service.readMany(savedKeys, req.sanitizedQuery);
+				res.locals['payload'] = { data: items };
+			} else {
+				const item = await service.readOne(savedKeys[0]!, req.sanitizedQuery);
+				res.locals['payload'] = { data: item };
+			}
+		} catch (error: any) {
+			if (isDirectusError(error, ErrorCode.Forbidden)) {
+				return next();
+			}
+
+			throw error;
+		}
+
+		const extensionManager = await getExtensionManager();
+		await extensionManager.reloadExtensionPermissions();
+
+		return next();
+	}),
+	respond
+);
+
+const readHandler = asyncHandler(async (req, res, next) => {
```

Comment:
Unrelated to this PR, but man we should consolidate all this boilerplate into a single "create standard controller"